packages/cicero-server/app.js

```diff
@@ -162,14 +202,207 @@ app.post('/draft/:template', async function (req, httpResponse, next) {
     }
 });
 
+/**
+ * Handle POST requests to /instantiate/:template
+ * The body of the POST should contain the request data and any options.
+ * The contract instance is created using the template and the data and saved under $CICERO_CONTRACTS path.
+ * The call returns an object having the data, instatiator's name, author's signatures, party signatures and the history.
+ *
+ * Instance Name
+ * ----------
+ * The instanceName parameter is the name of a directory under CICERO_CONTRACTS that contains
+ * the contract to use.
+ *
+ * Request
+ * ----------
+ * The POST body contains three properties:
+ *  - data
+ *  - instantiator
+ *  - output (optional - for saving updated contract instance in a new slc file under CICERO_CONTRACTS
+ *  - options
+ *
+ * Response
+ * ----------
+ * An object containing the data, instatiator's name, author's signatures, party signatures and the history
+ *
+ */
+app.post('/instantiate/:instanceName', async function (req, httpResponse, next) {
```

Comment:
I'm finding these "polymorphic" rest routes confusing. Would it not be easier to have something like:

`/templates/foo/contract/bar/trigger` to trigger a specific contract, or `/contracts/bar/trigger` if we don't want to nest contracts under templates?

Reply:
Agreed.  Would be good to have different end points for templates vs. contract instances.